frontend/src/components/organisms/ManageUsers.tsx

```diff
@@ -155,12 +160,19 @@ const ManageUsers = ({}: ManageUsersProps) => {
   let cursor = "";
 
   /** If a valid cursor is passed, fetch the next batch of users */
-  const fetchUsersBatch = async (cursor?: string) => {
+  const fetchUsersBatch = async (cursor?: string, searchQuery?: string) => {
     if (cursor !== "") {
-      const { response, data } = await api.get(
-        `/users?limit=${paginationModel.pageSize}&after=${cursor}`
-      );
-      return data;
+      if (searchQuery) {
+        const { response, data } = await api.get(
+          `/users?firstName=${searchQuery}`
```

Comment:
One thing I noticed is that the searching is case-sensitive. Please take a look at this: https://www.prisma.io/docs/orm/prisma-client/queries/case-sensitivity#:~:text=%7D)-,Options%20for%20case%2Dinsensitive%20filtering,-The%20recommended%20way

to see how case insensitive filtering would work in Prisma